configs/base.json

```diff
@@ -0,0 +1,12 @@
+{
+    "compilerOptions": {
+        "esModuleInterop": true,
+        "forceConsistentCasingInFileNames": true,
+        "noImplicitOverride": true,
+        "noImplicitReturns": true,
+        "noPropertyAccessFromIndexSignature": true,
+        "noUncheckedIndexedAccess": true,
+        "skipLibCheck": true,
```

Comment:
[`skipLibCheck`](https://www.typescriptlang.org/tsconfig#skipLibCheck) is the only one that isn't a slam dunk for me. It turns of type checking of .d.ts files in node_modules as a performance optimization, which can cause problems if you pull in two different versions of the same library.

Do we have a sense for how common this problem is or how much this improves performance? In the absence of any data, I'd vote to keep stricter checks (i.e. `"skipLibCheck": false`) but could be swayed if we have experience with this.

Reply:
I wouldn't think pulling two different versions of a library is common for front-end projects—maybe Node.js? Doesn't npm install conflicts as a subdirectory of the dependent?

Reply:
We discussed in person. There was [some concern](https://stackoverflow.com/questions/66817188/how-to-use-declare-an-external-module-which-wont-pass-strict-and-noimplicitr) that setting this to `false` would cause our strict tsconfig to start reporting errors on type declarations of libraries that use looser compiler configurations. This would be problematic because there wouldn't be ways to work around it, especially if your app wasn't even using the signature that was weakly typed. 

@mure  could you experiment and see if our understanding of the behavior is correct?

Reply:
I did some experiments and wasn't able to find any ways that having `skipLibCheck: false` would cause problems importing libraries with looser config options. We decided in person that we should go ahead and keep it false. We have anecdotal evidence that having it off doesn't cause any noticeable performance issues (from the nimble repo).